packages/signers/src/lit-protocol/signer.ts

```diff
@@ -0,0 +1,197 @@
+import { PKPEthersWallet } from "@lit-protocol/pkp-ethers";
+import { LitAbility, LitPKPResource } from "@lit-protocol/auth-helpers";
+import {
+  type AuthCallbackParams,
+  type AuthSig,
+  type SessionSigsMap,
+} from "@lit-protocol/types";
+import { LitNodeClient } from "@lit-protocol/lit-node-client";
+import { generateSessionKeyPair } from "@lit-protocol/crypto";
+
+import { type SignTypedDataParams } from "@alchemy/aa-core";
+import type { Address, TypedDataDomain } from "viem";
+import {
+  type LitAuthMethod,
+  type LitSessionSigsMap,
+  type LitConfig,
+  type LitAuthenticateProps,
+  type LitSmartAccountAuthenticator,
+  type LitUserMetadata,
+} from "./types.js";
+
+const SIGNER_TYPE: string = "lit";
+
+/**
+ * Implementation of `SmartAccountAuthenticator` for lit protocol
+ * This class requies:
+ * `@lit-protocol/lit-node-client@cayenne`
+ * `@lit-protocol/pkp-ethers@cayenne`
+ * `@lit-protocol/crypto@cayenne`
+ * `@lit-protocol/auth-helpers@cayenne`
+ * `@lit-protocol/types@cayenne`
+ */
+export class LitSigner<C extends LitAuthMethod | LitSessionSigsMap>
+  implements LitSmartAccountAuthenticator<C>
+{
+  inner: LitNodeClient;
+  public signer: PKPEthersWallet | undefined;
+  private _pkpPublicKey: string;
+  private _rpcUrl: string;
+  private _authContext: C | undefined;
+  public session: SessionSigsMap | undefined;
+
+  constructor(params: LitConfig) {
+    this._pkpPublicKey = params.pkpPublicKey;
+    this.inner =
+      params.inner ??
+      new LitNodeClient({
+        litNetwork: params.network ?? "cayenne",
+        debug: params.debug ?? false,
+      });
+    this._rpcUrl = params.rpcUrl;
+  }
+  signerType: string = SIGNER_TYPE;
+
+  /**
+   * if generic type is `LitAuthMethod`, authenticates the supplied authentication material.
+   * if type `SessionSigsMap`, this implementation will respect the existing auth and use the session material.
+   * @param props {LITAuthenticateProps} Authentication params, only `context` is required
+   * @returns {LitSessionSigsMap} Authenticated session material
+   */
+  authenticate = async (
+    props: LitAuthenticateProps<C>
+  ): Promise<LitUserMetadata> => {
+    if (!this.session) {
+      await this._doAuthentication(props);
+    }
+    if (!this.session) {
+      throw new Error("Not Authenticated");
+    }
+
+    return this.session;
+  };
+
+  getAuthDetails = async (): Promise<LitUserMetadata> => {
+    this._checkInternals();
+    return this._authContext as LitSessionSigsMap;
+  };
+
+  getAddress = async () => {
+    this._checkInternals();
+    const address = await this.signer?.getAddress();
+
+    return address as `0x${string}`;
```

Comment:
no need to await since we're returning a promise anyways. also you can use the viem Address type
```suggestion
    return this.signer?.getAddress() as Promise<Address>;
```